File: threads/src/main/java/im/threads/internal/transport/models/AttachmentSettings.java
package im.threads.internal.transport.models;

public class AttachmentSettings {

    private String clientId;
    private Content content;

    public String getClientId() {
        return clientId;
    }

    public Content getContent() {
        return content;
    }

    public static class Content {
        private int maxSize;
        private String[] fileExtensions;

        public Content() {
        }

        public Content(int maxSize, String[] fileExtensions) {
            this.maxSize = maxSize;
            this.fileExtensions = fileExtensions;
        }

        public int getMaxSize() {
            return maxSize;
        }

        public String[] getFileExtensions() {
            return fileExtensions;
        }
    }
}

File: src/main/java/im/threads/business/transport/models/AttachmentSettings.kt
package im.threads.business.transport.models

class AttachmentSettings {
    var clientId: String? = null
    var content: Content? = null

    class Content(maxSize: Int, fileExtensions: Array<String>) {
        var maxSize = maxSize
            private set
        var fileExtensions: Array<String> = fileExtensions
            private set
    }
}
